### ISSUE TITLE
Reviewing​ names of languages

### ISSUE THREAD

**Issue by trang** [Tatoeba/tatoeba2#936]
There has been requests to rename some languages such as ["Moroccan Arabic" into "Arabic (Moroccan)"](https://tatoeba.org/eng/wall/show_message/24920#message_24920), or ["Literary Chinese" into "Chinese (literary)"](https://github.com/Tatoeba/tatoeba2/issues/750)

There are many cases like these​ ​where it may make sense to rename the language so that it can be found more easily by users. It is however not always obvious.

​There is also the case of a language (Odia) which name given by the SIL is outdated, and we decided to use the more updated name, leaving into parenthesis the outdated one.

In order to make the naming consistent, we need to review the names of all the languages.​

---

Below is the list of languages for which we altered the name.

Chinese (Mandarin)
Norwegian (Bokmål)
Norwegian (Nynorsk)
Punjabi (Eastern)
Punjabi (Western)
Juhuri (Judeo-Tat)
Odia (Oriya)

---

​Below is the list of the languages which names are a combinations of words.

**Which of these languages should we rename?**

Low Saxon
Literary Chinese
Old East Slavic
Central Dusun
Old Prussian
Upper Sorbian
Lower Sorbian
Algerian Arabic
Scottish Gaelic
Egyptian Arabic
North Moluccan Malay
Coastal Kadazan
Crimean Tatar
Ancient Greek
Ottoman Turkish
Old English
Hill Mari
Tok Pisin
Haitian Creole
Guerrero Nahuatl
Nigerian Fulfulde
Fiji Hindi
Lingua Franca Nova
North Levantine Arabic
Pennsylvania German
Northern Sami
Orizaba Nahuatl
Middle English
Middle French
Min Nan Chinese
Southern Altai
Palatine German
Old Norse
Tagal Murut
Old Aramaic
Hakka Chinese
Southern Sotho
Swiss German
Louisiana Creole
Moroccan Arabic
Old Saxon
Seychellois Creole

---

​There may be languages that are not in the list above but which should still be renamed, for reasons similar to Odia.

**Comment by jiru** [comment]
As far as I understand, the problem with multiwords language names is that they may not sticking together in the dropdown list and are hard to find. However, names with parenthesis (or otherwise long) tends not to be appropriate when included into sentences. For instance, the title of [this page](https://tatoeba.org/eng/sentences/show_all_in/cmn/none/nob/indifferent) is `Sentences in Chinese (Mandarin) not translated into Norwegian (Bokmål) (48486 results)` is not very appropriate.

Maybe it would be a good idea to keep two names: one “extended” that defines extensively the language, appropriate for lists, and one “short” that is commonly used when referring to it, appropriate for everywhere else.

We may also mitigate the list problem by completing the unfinished advanced language selector.

**Comment by trang** [comment]
> However, names with parenthesis (or otherwise long) tends not to be appropriate when included into sentences.

That's a good point.

**Comment by RyckRichards** [comment]
https://tatoeba.org/eng/wall/index/26031#message_26031

> ** Renaming Arabic languages ** (for now)

cueyayotl told me that many of our Arabic users are not finding their varieties of Arabic - even when they exist. He proposed to change all of them to the following:

Arabic (Modern Standard)
Arabic (Algerian)
Arabic (Egyptian)
Arabic (Iraqi)
Arabic (Moroccan)
Arabic (North Levantine)


We have an open issue on GitHub about that already - https://github.com/Tatoeba/tatoeba2/issues/936 . Do you agree with cueyayotl's suggestion for this case or not? I'd like to hear from you.

**Comment by sabretou** [comment]
Cueyayotl's idea has also been mine own for a while now, which is why, when Punjabi was to be split into Eastern and Western, I specifically recommended it be split with parentheses so that both forms have easy visibility. That said, we can change the names on a case by case basis. If too many new users are adding sentences under the wrong language, we can address that language.

At present, Cueyayotl has confirmed that Arabic language names _are_ causing problems, so I would suggest we change those first. The question we must ask is, "What language name will a speaker of that language search?" In Punjabi's case, for example, a speaker will not search for Eastern or Western Punjabi, only Punjabi. Arabic, I understand, would be the same way. I think a similar situation would be with with Nahuatl and Chinese, so those varieties could be next.

**Comment by trang** [comment]
I think working on a better language dropdown would be the best solution. For languages like Arabic, the problem is not the name itself, it's just the fact that users can't find the name.
There are languages for which it will be impossible for us to evaluate what part of the name users will search first. Changing the names might be a short term solution for some languages, but in the long term we will need a better language dropdown. And once we have a  better dropdown, we will have to change back all the names... So we might as well just focus on implementing a better dropdown rather than changing names back and forth.

**Comment by RyckRichards** [comment]
@cueyayotl are there other languages that should be renamed?

**Comment by trang** [comment]
I was hesitating whether or not to merge @RyckRichards's PR about renaming Chinese languages. I decided not to.

I have created an issue to make a searchable dropdown for language. This should fix the problem of people not finding their language even though it exists. For instance if someone types "ara", they would see all languages which name contains "ara".
I'm aiming to implement this dropdown in the next couple of weeks so that we can finally make our language names consistent.

For this particular issue:
1. We should fix the language names that we have altered, so that they comply to the name used in the ISO 639-3 definition. Only exception would be `Odia (Oriya)`. We keep this one as is (for now at least), otherwise people wouldn't be able to find the language if they looked it up with the old name.
2. If there's any new language request from now on, we should always keep the original name. We shouldn't alter it with parenthesis.
